hugegraph-core/src/main/java/com/baidu/hugegraph/traversal/algorithm/ShortestPathTraverserNew.java

```diff
@@ -0,0 +1,187 @@
+import com.baidu.hugegraph.structure.HugeEdge;
+import com.baidu.hugegraph.type.define.Directions;
+import com.baidu.hugegraph.util.E;
+import com.google.common.collect.ImmutableList;
+
+public class ShortestPathTraverserNew extends HugeTraverser {
+
+    public ShortestPathTraverserNew(HugeGraph graph) {
+        super(graph);
+    }
+
+    public List<Id> shortestPath(Id sourceV, Id targetV, Directions dir,
+                                 String label, int depth,int max_edges, long degree,
+                                 long capacity) {
+        E.checkNotNull(sourceV, "source vertex id");
+        E.checkNotNull(targetV, "target vertex id");
+        E.checkNotNull(dir, "direction");
+        checkPositive(depth, "max depth");
+        checkDegree(degree);
+        checkCapacity(capacity);
+
+        if (sourceV.equals(targetV)) {
+            return ImmutableList.of(sourceV);
+        }
+
+        Id labelId = this.getEdgeLabelId(label);
+        Traverser traverser = new Traverser(sourceV, targetV, dir, labelId,max_edges,
+                                            degree, capacity);
+        List<Id> path;
+        while (true) {
+            // Found, reach max depth or reach capacity, stop searching
+            if ((path = traverser.forward()) != PATH_NONE || --depth <= 0) {
+                break;
+            }
+            checkCapacity(traverser.capacity, traverser.size, "shortest path");
+
+            if ((path = traverser.backward()) != PATH_NONE || --depth <= 0) {
+                Collections.reverse(path);
+                break;
+            }
+            checkCapacity(traverser.capacity, traverser.size, "shortest path");
+        }
+        return path;
+    }
+
+    private class Traverser {
+
+        // TODO: change Map to Set to reduce memory cost
+        private Map<Id, Node> sources = newMap();
+        private Map<Id, Node> targets = newMap();
+
+        private final Directions direction;
+        private final Id label;
+        private final long degree;
+        private final long capacity;
+        private long size;
+        private final int max_edges;
+
+        public Traverser(Id sourceV, Id targetV, Directions dir,
+                         Id label,int max_edges, long degree, long capacity) {
+            this.sources.put(sourceV, new Node(sourceV));
+            this.targets.put(targetV, new Node(targetV));
+            this.direction = dir;
+            this.label = label;
+            this.degree = degree;
+            this.capacity = capacity;
+            this.size = 0L;
+            this.max_edges=max_edges;
+        }
+
+        /**
+         * Search forward from source
+         */
+        public List<Id> forward() {
+            Map<Id, Node> newVertices = newMap();
+            // Traversal vertices of previous level
+            for (Node v : this.sources.values()) {
+                //TODO 判断是否为超级节点，如当前节点边的数量大于1000，就不经过滤当前点
+                if(v.path().size()< max_edges) {
```

Comment:
'v.path()' returns node list traversed from source to current 'v'.
Should limit one node's edges number to avoid super node.